lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -84,24 +84,30 @@ def add_node_data(node)
   end
 
   # Rewrite a given file resource with the metadata from a fileserver based file
-  def replace_metadata(resource, metadata, recurse = false)
+  def replace_metadata(resource, metadata)
     if resource[:links] == "manage" && metadata.ftype == "link"
       resource.delete(:source)
-      resource[:ensure] = "link"
       resource[:target] = metadata.destination
-    elsif resource[:links] == "follow"
-      resource[:ensure] = metadata.ftype
     end
 
+    resource[:ensure] = metadata.ftype
```

Comment:
The 3 values for ftype are `link`, `file`, and `directory`. Previously we updated if `links => manage` or `links => follow`; the third option is `links => ignore`.

The cases where this would now update, but hadn't before:
- request a link but get a file or directory
- request a file but get a directory
- request a directory but get a file

The `#copy_from_source` method in the `source` parameter would do those things. If we don't do them in the static catalog, I'm not sure we get correct behavior, since we skip `#copy_from_source` if `checksum_value` is set. I need to investigate.